### DISCUSSION TITLE
Extract Rack metadata middleware from RailsEventStore gem

### DISCUSSION THREAD

**Discussion by mostlyobvious** [RailsEventStore/rails_event_store#1900] [Ideas]
It is possible to use Rack middleware that enhances recorded events metadata outside of Rails. Yet we ship it in the Rails version of the gem (`rails_event_store`): https://github.com/RailsEventStore/rails_event_store/blob/409339adbe094dabd439e15e46daaf145be7f647/rails_event_store/lib/rails_event_store/middleware.rb

A wider audience could benefit from it if that was present in `ruby_event_store`. Still, `rails_event_store` may continue enabling it by default: 
https://github.com/RailsEventStore/rails_event_store/blob/409339adbe094dabd439e15e46daaf145be7f647/rails_event_store/lib/rails_event_store/middleware.rb

**Comment by joelvh** [comment]
What are your thoughts about putting much more into “contrib”? Things like Rack and workers and other plumbing to integrate in many ways? All the modular “plugins”, essentially. I saw your other ticket about documenting how to use RES with Hanami and Sidekiq, and also to separate AR from Rails — put all the pieces in contrib and then have a place that helps pull the pieces together for different integrations, etc.

**Comment by mostlyobvious** [comment]
@joelvh Feel free to put more intro `contrib/`. So far the common thing about it is that code there does  not get released along "official" RES gems and does not necessarily has to follow 100% mutation coverage nor versioning rules.

I'd be glad to see there more stuff ✨

**Comment by mostlyobvious** [comment]
Related: https://www.loom.com/share/49caaa09ded0452db024f199e20eb9ae
